mantle/cmd/kola/testiso.go

```diff
@@ -331,50 +416,36 @@ func runTestIso(cmd *cobra.Command, args []string) error {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	targetScenarios := make(map[string]bool)
-	for _, scenario := range scenarios {
-		if _, ok := allScenarios[scenario]; !ok {
-			return fmt.Errorf("Unknown scenario: %s", scenario)
-		}
-		targetScenarios[scenario] = true
-	}
-
-	// Only x86_64 supports hybrid ISOs.
-	if coreosarch.CurrentRpmArch() != "x86_64" {
-		fmt.Printf("Skipping iso-as-disk; not supported on %s\n", coreosarch.CurrentRpmArch())
-		delete(targetScenarios, scenarioISOAsDisk)
-	}
-
-	// s390x: iso-install does not work because s390x uses an El Torito image
-	if coreosarch.CurrentRpmArch() == "s390x" {
-		fmt.Println("Skipping iso-install on s390x")
-		noiso = true
+	for _, s := range  archPatterns  {
+		allPatterns[s] = true
 	}
-
-	if nopxe {
-		delete(targetScenarios, scenarioPXEInstall)
-		delete(targetScenarios, scenarioPXEOfflineInstall)
-	}
-	if noiso {
-		delete(targetScenarios, scenarioISOInstall)
-		delete(targetScenarios, scenarioISOOfflineInstall)
-		delete(targetScenarios, scenarioMinISOInstall)
-		delete(targetScenarios, scenarioMinISOInstallNm)
-		delete(targetScenarios, scenarioISOLiveLogin)
+	targetPatterns := make(map[string]bool)
+	for _, pattern := range patterns {
+		if _, ok := allPatterns[pattern]; !ok {
+			return fmt.Errorf("Unknown pattern: %s", pattern)
+		}
+		targetPatterns[pattern] = true
 	}
 
 	// just make it a normal print message so pipelines don't error out for ppc64le and s390x
-	if len(targetScenarios) == 0 {
-		fmt.Println("No valid scenarios specified!")
+	if len(targetPatterns) == 0 {
+		fmt.Println("No valid patterns specified!")
 		return nil
 	}
```

Comment:
What do you mean? `targetPatterns` is the result of the check for `patterns`

Reply:
I think what I'm trying to say is that the previous for loop errors if any patterns aren't valid so by the time we get here `targetPatterns.keys()` should be the same as `patterns` (I think). So we might as well just check if `len(patterns) == 0` up front.

Is my understanding wrong?

Reply:
Ah ok, actually I think we don't need this check at all. If we don't pass anything it will run the default arch list, if some test is wrong it will error out.

Reply:
sounds good to me